File: PiecewiseConstant/synthesize/plots/environment.m
%% Plotting pendulum grid
clc; clear; close all

% Load data files
data_hyper = load('../models/pendulum/partition_data_120.mat');

% Define state space
partitions = data_hyper.partitions;
initial_state = 65;
obstacle_state = false;

% Plot the grid
plot_env(partitions, initial_state, obstacle_state)

%% Functions
function plots = plot_env(partitions, initial_state, obstacle_state)

    figure
    
    for jj = 1:length(partitions)

        parts = partitions(jj, :, :);

        x_1_low = parts(1);
        x_1_up = parts(2);

        x_2_low = parts(3);
        x_2_up = parts(4);

        P = [x_1_low x_2_low; x_1_low x_2_up; x_1_up x_2_up; x_1_up x_2_low];

        if jj == initial_state
            plots = patch(P(:,1),P(:,2), 0.8, 'EdgeAlpha', 0.1);
        elseif jj == obstacle_state
            plots = patch(P(:,1),P(:,2), 0.2, 'EdgeAlpha', 0.1);
        else
            plots = patch(P(:,1),P(:,2), 0.6, 'EdgeAlpha', 0.1);
        end

        
    end
    
    % Plot properties
    c = colormap(jet(256));
    shading faceted
    caxis([0 1])
    colormap(flipud(c));

    % Label
    xlabel("$\theta$ (rad)", 'Interpreter','latex', "FontSize", 25)
    ylabel('$\dot{\theta}$ (rad/s)', 'Interpreter','latex', "FontSize", 25)
    set(gcf,'color','w');
    set(gca,'FontSize',20)
    
end

File: PiecewiseConstant/synthesize/plots/environment.jl
using Plots, MAT
plotlyjs()

function plot_env(partitions, initial_state)
    # Define custom colors for different regions
    colors = ["#FF0000", "#00FF00", "#0000FF"]
    
    # Create a new plot
    p = plot(legend = false)
    
    for jj in eachindex(partitions[:, 1, 1])
        parts = partitions[jj, :, :]
        x_1_low = parts[1]
        x_1_up = parts[2]
        x_2_low = parts[3]
        x_2_up = parts[4]
        
        P = [x_1_low x_2_low;
             x_1_low x_2_up;
             x_1_up  x_2_up;
             x_1_up  x_2_low]
        
        if jj in initial_state
            plot!(P[:, 1], P[:, 2], alpha=0.8, color=colors[1])
        elseif jj == obstacle_state
            plot!(P[:, 1], P[:, 2], alpha=0.2, color=colors[2])
        else
            plot!(P[:, 1], P[:, 2], alpha=0.6, color=colors[3])
        end
    end
    
    # Plot properties
    xlabel!("θ (rad)")
    ylabel!("θ̇ (rad/s)")
    title!("Pendulum Grid")
    xticks!(0:20:120)
    yticks!(0:20:120)
    xgrid!
    ygrid!
    # legend = false

    display(p)
end

# Load data files
data_hyper = matread("../models/pendulum/partition_data_120.mat")

# Define state space
partitions = data_hyper["partitions"]
initial_state = [55, 56, 65, 66]

# Plot the grid
plot_env(partitions, initial_state)
